### ISSUE TITLE
Semantic highlighting errors

### ISSUE THREAD

**Issue by rietmann** [cquery-project/emacs-cquery#5]
Trying the melpa version (`20180121.2203`), I get the following errors:

    error in process filter: cquery--publish-semantic-highlighting: Symbol’s function definition is void: face
    error in process filter: Symbol’s function definition is void: face

I'm using spacemacs, trying the lsp layer in [this pull request](https://github.com/syl20bnr/spacemacs/pull/10211), and a cquery layer that looks like:

```elisp
(defconst cquery-packages
  '(
    ;; lsp-mode ;; lsp required via (configuration-layer/declare-layer 'lsp)
    company-lsp
    cquery
    ))

;; (defun cquery/init-lsp-mode ()
;;   (use-package lsp-mode))

(defun cquery/post-init-company-lsp ()
  (message "trying adding company-lsp to c-mode-common backends")
  (push 'company-lsp company-backends-c-mode-common)
  )

(defun cquery/init-cquery ()
  (use-package cquery
    :init
    (progn
      (spacemacs/add-to-hooks #'lsp-cquery-enable '(c-mode-hook c++-mode-hook))
      )))
```

And a few tweaks in `user-config`
```elisp
(setq cquery-additional-arguments (list "--log-file" "cquery.log"))
  (setq cquery-executable "/home/rietmann/.local/stow/cquery/bin/cquery")
  ;; (setq lsp-highlight-symbol-at-point nil)
  ;; (face-spec-set 'lsp-face-highlight-textual '((t :background nil :inherit hl-line)))
  (setq lsp-ui-sideline-show-hover nil)
```

I'm not too good at debugging elisp, but I thought you should know I'm having trouble and my config.

**Comment by rietmann** [comment]
Debugger entered--Lisp error: (void-function face)
  face(cquery-sem-free-var-face)
  cquery--publish-semantic-highlighting([cl-struct-lsp--workspace [cl-struct-lsp--parser nil nil nil nil nil nil nil "" nil nil #0] #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("/home/rietmann/drz/c++_test/src/Main.cpp" 0 "/home/rietmann/drz/c++_test/src/testquick.cpp" 157 ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("textDocumentSync" 2 "hoverProvider" t "completionProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil "triggerCharacters" ("." ":" ">" "#") ...)) "signatureHelpProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("triggerCharacters" ("(" ",") ...)) "definitionProvider" t "referencesProvider" t "documentHighlightProvider" t "documentSymbolProvider" t "workspaceSymbolProvider" t "codeActionProvider" t "codeLensProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil ...)) "documentFormattingProvider" nil "documentRangeFormattingProvider" nil "renameProvider" t "documentLinkProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil ...)) ...)) nil "/home/rietmann/drz/c++_test/" [cl-struct-lsp--client #[257 "\300\207" ["cpp"] 2 "cpp\n\n(fn _)"] nil nil nil #[514 "\302\203	\302 \202\n\301\211:\203\211\202\211C\304@!\204$\305\306\307@\"!\210\310\311\300\312\313\314\315\316\317
\320\321\303\322\323&\324\325\"\210\324\326\327\303!!\325\"\210\211\262\207" ["lsp-cquery" ("/home/rietmann/.local/stow/cquery/bin/cquery" "--language-server" "--log-file" "cquery.log") nil "*lsp-cquery stderr*" executable-find error format "Couldn't find executable %s" make-process :name :connection-type pipe :coding no-conversion :command :filter :sentinel :stderr :noquery t set-process-query-on-exit-flag nil get-buffer-process get-buffer] 21 "\n\n(fn FILTER SENTINEL)"] "*lsp-cquery stderr*" cquery--get-root nil #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("$cquery/setInactiveRegions" (lambda (w p) (cquery--set-inactive-regions w p)) "$cquery/publishSemanticHighlighting" (lambda (w p) (cquery--publish-semantic-highlighting w p)) "$cquery/progress" (lambda (_w _p)) ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ( ...)) #s(hash-table size 65 test eq rehash-size 1.5 rehash-threshold 0.8 data ( ...)) (("objectivec" lambda (str) (with-temp-buffer (delay-mode-hooks (objc-mode)) (insert str) (font-lock-ensure) (buffer-string))) ("cpp" lambda (str) (with-temp-buffer (delay-mode-hooks (c++-mode)) (insert str) (font-lock-ensure) (buffer-string))) ("c" lambda (str) (with-temp-buffer (delay-mode-hooks (c-mode)) (insert str) (font-lock-ensure) (buffer-string)))) 71] nil #<process lsp-cquery> #<process lsp-cquery> (#<buffer testquick.cpp> #<buffer Main.cpp>) nil ((company-lsp . company-lsp--client-capabilities))] #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("uri" "file:///home/rietmann/drz/c++_test/src/testquick.cpp" "symbols" (#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("type" 2 "kind" 13 "isTypeMember" nil "stableId" 15 "ranges" (#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("start" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 4 "character" 6 ...)) "end" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 4 "character" 11 ...)) ...))) ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("type" 1 "kind" 12 "isTypeMember" nil "stableId" 14 "ranges" (#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("start" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 2 "character" 5 ...)) "end" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 2 "character" 9 ...)) ...))) ...))) ...)))
  (lambda (w p) (cquery--publish-semantic-highlighting w p))([cl-struct-lsp--workspace [cl-struct-lsp--parser nil nil nil nil nil nil nil "" nil nil #0] #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("/home/rietmann/drz/c++_test/src/Main.cpp" 0 "/home/rietmann/drz/c++_test/src/testquick.cpp" 157 ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("textDocumentSync" 2 "hoverProvider" t "completionProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil "triggerCharacters" ("." ":" ">" "#") ...)) "signatureHelpProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("triggerCharacters" ("(" ",") ...)) "definitionProvider" t "referencesProvider" t "documentHighlightProvider" t "documentSymbolProvider" t "workspaceSymbolProvider" t "codeActionProvider" t "codeLensProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil ...)) "documentFormattingProvider" nil "documentRangeFormattingProvider" nil "renameProvider" t "documentLinkProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil ...)) ...)) nil "/home/rietmann/drz/c++_test/" [cl-struct-lsp--client #[257 "\300\207" ["cpp"] 2 "cpp\n\n(fn _)"] nil nil nil #[514 "\302\203	\302 \202\n\301\211:\203\211\202\211C\304@!\204$\305\306\307@\"!\210\310\311\300\312\313\314\315\316\317
\320\321\303\322\323&\324\325\"\210\324\326\327\303!!\325\"\210\211\262\207" ["lsp-cquery" ("/home/rietmann/.local/stow/cquery/bin/cquery" "--language-server" "--log-file" "cquery.log") nil "*lsp-cquery stderr*" executable-find error format "Couldn't find executable %s" make-process :name :connection-type pipe :coding no-conversion :command :filter :sentinel :stderr :noquery t set-process-query-on-exit-flag nil get-buffer-process get-buffer] 21 "\n\n(fn FILTER SENTINEL)"] "*lsp-cquery stderr*" cquery--get-root nil #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("$cquery/setInactiveRegions" (lambda (w p) (cquery--set-inactive-regions w p)) "$cquery/publishSemanticHighlighting" (lambda (w p) (cquery--publish-semantic-highlighting w p)) "$cquery/progress" (lambda (_w _p)) ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ( ...)) #s(hash-table size 65 test eq rehash-size 1.5 rehash-threshold 0.8 data ( ...)) (("objectivec" lambda (str) (with-temp-buffer (delay-mode-hooks (objc-mode)) (insert str) (font-lock-ensure) (buffer-string))) ("cpp" lambda (str) (with-temp-buffer (delay-mode-hooks (c++-mode)) (insert str) (font-lock-ensure) (buffer-string))) ("c" lambda (str) (with-temp-buffer (delay-mode-hooks (c-mode)) (insert str) (font-lock-ensure) (buffer-string)))) 71] nil #<process lsp-cquery> #<process lsp-cquery> (#<buffer testquick.cpp> #<buffer Main.cpp>) nil ((company-lsp . company-lsp--client-capabilities))] #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("uri" "file:///home/rietmann/drz/c++_test/src/testquick.cpp" "symbols" (#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("type" 2 "kind" 13 "isTypeMember" nil "stableId" 15 "ranges" (#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("start" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 4 "character" 6 ...)) "end" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 4 "character" 11 ...)) ...))) ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("type" 1 "kind" 12 "isTypeMember" nil "stableId" 14 "ranges" (#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("start" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 2 "character" 5 ...)) "end" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 2 "character" 9 ...)) ...))) ...))) ...)))
  lsp--on-notification([cl-struct-lsp--parser nil nil nil nil nil nil nil "" nil nil [cl-struct-lsp--workspace #0 #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("/home/rietmann/drz/c++_test/src/Main.cpp" 0 "/home/rietmann/drz/c++_test/src/testquick.cpp" 157 ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("textDocumentSync" 2 "hoverProvider" t "completionProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil "triggerCharacters" ("." ":" ">" "#") ...)) "signatureHelpProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("triggerCharacters" ("(" ",") ...)) "definitionProvider" t "referencesProvider" t "documentHighlightProvider" t "documentSymbolProvider" t "workspaceSymbolProvider" t "codeActionProvider" t "codeLensProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil ...)) "documentFormattingProvider" nil "documentRangeFormattingProvider" nil "renameProvider" t "documentLinkProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil ...)) ...)) nil "/home/rietmann/drz/c++_test/" [cl-struct-lsp--client #[257 "\300\207" ["cpp"] 2 "cpp\n\n(fn _)"] nil nil nil #[514 "\302\203	\302 \202\n\301\211:\203\211\202\211C\304@!\204$\305\306\307@\"!\210\310\311\300\312\313\314\315\316\317
\320\321\303\322\323&\324\325\"\210\324\326\327\303!!\325\"\210\211\262\207" ["lsp-cquery" ("/home/rietmann/.local/stow/cquery/bin/cquery" "--language-server" "--log-file" "cquery.log") nil "*lsp-cquery stderr*" executable-find error format "Couldn't find executable %s" make-process :name :connection-type pipe :coding no-conversion :command :filter :sentinel :stderr :noquery t set-process-query-on-exit-flag nil get-buffer-process get-buffer] 21 "\n\n(fn FILTER SENTINEL)"] "*lsp-cquery stderr*" cquery--get-root nil #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("$cquery/setInactiveRegions" (lambda (w p) (cquery--set-inactive-regions w p)) "$cquery/publishSemanticHighlighting" (lambda (w p) (cquery--publish-semantic-highlighting w p)) "$cquery/progress" (lambda (_w _p)) ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ( ...)) #s(hash-table size 65 test eq rehash-size 1.5 rehash-threshold 0.8 data ( ...)) (("objectivec" lambda (str) (with-temp-buffer (delay-mode-hooks (objc-mode)) (insert str) (font-lock-ensure) (buffer-string))) ("cpp" lambda (str) (with-temp-buffer (delay-mode-hooks (c++-mode)) (insert str) (font-lock-ensure) (buffer-string))) ("c" lambda (str) (with-temp-buffer (delay-mode-hooks (c-mode)) (insert str) (font-lock-ensure) (buffer-string)))) 71] nil #<process lsp-cquery> #<process lsp-cquery> (#<buffer testquick.cpp> #<buffer Main.cpp>) nil ((company-lsp . company-lsp--client-capabilities))]] #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("jsonrpc" "2.0" "method" "$cquery/publishSemanticHighlighting" "params" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("uri" "file:///home/rietmann/drz/c++_test/src/testquick.cpp" "symbols" (#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("type" 2 "kind" 13 "isTypeMember" nil "stableId" 15 "ranges" (#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("start" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 4 "character" 6 ...)) "end" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 4 "character" 11 ...)) ...))) ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("type" 1 "kind" 12 "isTypeMember" nil "stableId" 14 "ranges" (#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("start" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 2 "character" 5 ...)) "end" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("line" 2 "character" 9 ...)) ...))) ...))) ...)) ...)))
  lsp--parser-on-message([cl-struct-lsp--parser nil nil nil nil nil nil nil "" nil nil [cl-struct-lsp--workspace #0 #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("/home/rietmann/drz/c++_test/src/Main.cpp" 0 "/home/rietmann/drz/c++_test/src/testquick.cpp" 157 ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("textDocumentSync" 2 "hoverProvider" t "completionProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil "triggerCharacters" ("." ":" ">" "#") ...)) "signatureHelpProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("triggerCharacters" ("(" ",") ...)) "definitionProvider" t "referencesProvider" t "documentHighlightProvider" t "documentSymbolProvider" t "workspaceSymbolProvider" t "codeActionProvider" t "codeLensProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil ...)) "documentFormattingProvider" nil "documentRangeFormattingProvider" nil "renameProvider" t "documentLinkProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil ...)) ...)) nil "/home/rietmann/drz/c++_test/" [cl-struct-lsp--client #[257 "\300\207" ["cpp"] 2 "cpp\n\n(fn _)"] nil nil nil #[514 "\302\203	\302 \202\n\301\211:\203\211\202\211C\304@!\204$\305\306\307@\"!\210\310\311\300\312\313\314\315\316\317
\320\321\303\322\323&\324\325\"\210\324\326\327\303!!\325\"\210\211\262\207" ["lsp-cquery" ("/home/rietmann/.local/stow/cquery/bin/cquery" "--language-server" "--log-file" "cquery.log") nil "*lsp-cquery stderr*" executable-find error format "Couldn't find executable %s" make-process :name :connection-type pipe :coding no-conversion :command :filter :sentinel :stderr :noquery t set-process-query-on-exit-flag nil get-buffer-process get-buffer] 21 "\n\n(fn FILTER SENTINEL)"] "*lsp-cquery stderr*" cquery--get-root nil #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("$cquery/setInactiveRegions" (lambda (w p) (cquery--set-inactive-regions w p)) "$cquery/publishSemanticHighlighting" (lambda (w p) (cquery--publish-semantic-highlighting w p)) "$cquery/progress" (lambda (_w _p)) ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ( ...)) #s(hash-table size 65 test eq rehash-size 1.5 rehash-threshold 0.8 data ( ...)) (("objectivec" lambda (str) (with-temp-buffer (delay-mode-hooks (objc-mode)) (insert str) (font-lock-ensure) (buffer-string))) ("cpp" lambda (str) (with-temp-buffer (delay-mode-hooks (c++-mode)) (insert str) (font-lock-ensure) (buffer-string))) ("c" lambda (str) (with-temp-buffer (delay-mode-hooks (c-mode)) (insert str) (font-lock-ensure) (buffer-string)))) 71] nil #<process lsp-cquery> #<process lsp-cquery> (#<buffer testquick.cpp> #<buffer Main.cpp>) nil ((company-lsp . company-lsp--client-capabilities))]] "{\"jsonrpc\":\"2.0\",\"method\":\"$cquery/publishSemanticHighlighting\",\"params\":{\"uri\":\"file:///home/rietmann/drz/c++_test/src/testquick.cpp\",\"symbols\":[{\"type\":2,\"kind\":13,\"isTypeMember\":false,\"stableId\":15,\"ranges\":[{\"start\":{\"line\":4,\"character\":6},\"end\":{\"line\":4,\"character\":11}}]},{\"type\":1,\"kind\":12,\"isTypeMember\":false,\"stableId\":14,\"ranges\":[{\"start\":{\"line\":2,\"character\":5},\"end\":{\"line\":2,\"character\":9}}]}]}}")
  #[514 "\306\301\306\307\306:\203,@\262\310\"\203!\306\262\306\211\262\202\"\307\203,A\262\202\2034\307\2025\211\266\204\203\232\3111F\312\300\"0\202|\313\300!\210\300\314H>\204Y\315\316\317\300D\"\210\300\211\320\306I\266\300\314H>\204o\315\316\317\300D\"\210\300\211\321\306I\266\322\323\"\262\211\211\203\230\211@\f\203\214\324\325\"\210\326\300\"\210A\266\202\202}\266\300\314H>\204\251\315\316\317\300D\"\210\300\321H\205\302\3271\273\306\330!)0\207\210\307\331\332!\207" [[cl-struct-lsp--parser nil nil nil nil nil nil nil "" nil nil [cl-struct-lsp--workspace #2 #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("/home/rietmann/drz/c++_test/src/Main.cpp" 0 "/home/rietmann/drz/c++_test/src/testquick.cpp" 157 ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("textDocumentSync" 2 "hoverProvider" t "completionProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil "triggerCharacters" ("." ":" ">" "#") ...)) "signatureHelpProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("triggerCharacters" ("(" ",") ...)) "definitionProvider" t "referencesProvider" t "documentHighlightProvider" t "documentSymbolProvider" t "workspaceSymbolProvider" t "codeActionProvider" t "codeLensProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil ...)) "documentFormattingProvider" nil "documentRangeFormattingProvider" nil "renameProvider" t "documentLinkProvider" #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("resolveProvider" nil ...)) ...)) nil "/home/rietmann/drz/c++_test/" [cl-struct-lsp--client #[257 "\300\207" ["cpp"] 2 "cpp\n\n(fn _)"] nil nil nil #[514 "\302\203	\302 \202\n\301\211:\203\211\202\211C\304@!\204$\305\306\307@\"!\210\310\311\300\312\313\314\315\316\317
\320\321\303\322\323&\324\325\"\210\324\326\327\303!!\325\"\210\211\262\207" ["lsp-cquery" ("/home/rietmann/.local/stow/cquery/bin/cquery" "--language-server" "--log-file" "cquery.log") nil "*lsp-cquery stderr*" executable-find error format "Couldn't find executable %s" make-process :name :connection-type pipe :coding no-conversion :command :filter :sentinel :stderr :noquery t set-process-query-on-exit-flag nil get-buffer-process get-buffer] 21 "\n\n(fn FILTER SENTINEL)"] "*lsp-cquery stderr*" cquery--get-root nil #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ("$cquery/setInactiveRegions" (lambda (w p) (cquery--set-inactive-regions w p)) "$cquery/publishSemanticHighlighting" (lambda (w p) (cquery--publish-semantic-highlighting w p)) "$cquery/progress" (lambda (_w _p)) ...)) #s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8 data ( ...)) #s(hash-table size 65 test eq rehash-size 1.5 rehash-threshold 0.8 data ( ...)) (("objectivec" lambda (str) (with-temp-buffer ... ... ... ...)) ("cpp" lambda (str) (with-temp-buffer ... ... ... ...)) ("c" lambda (str) (with-temp-buffer ... ... ... ...))) 71] nil #<process lsp-cquery> #<process lsp-cquery> (#<buffer testquick.cpp> #<buffer Main.cpp>) nil ((company-lsp . company-lsp--client-capabilities))]] nil lsp--no-response cl-struct-lsp--parser-tags lsp-print-io inhibit-quit nil t string-match (error) lsp--parser-read lsp--parser-reset 0 signal wrong-type-argument lsp--parser 2 1 error "Error parsing language server output: %s" message "Output from language server: %s" lsp--parser-on-message (quit) accept-process-output eval (ignore nil) quit-flag] 9 "\n\n(fn PROC OUTPUT)"](#<process lsp-cquery> "Content-Length: 415
\n
\n{\"jsonrpc\":\"2.0\",\"method\":\"$cquery/publishSemanticHighlighting\",\"params\":{\"uri\":\"file:///home/rietmann/drz/c++_test/src/testquick.cpp\",\"symbols\":[{\"type\":2,\"kind\":13,\"isTypeMember\":false,\"stableId\":15,\"ranges\":[{\"start\":{\"line\":4,\"character\":6},\"end\":{\"line\":4,\"character\":11}}]},{\"type\":1,\"kind\":12,\"isTypeMember\":false,\"stableId\":14,\"ranges\":[{\"start\":{\"line\":2,\"character\":5},\"end\":{\"line\":2,\"character\":9}}]}]}}Content-Length: 179
\n
\n{\"jsonrpc\":\"2.0\",\"method\":\"$cquery/progress\",\"params\":{\"indexRequestCount\":0,\"doIdMapCount\":0,\"loadPreviousIndexCount\":0,\"onIdMappedCount\":0,\"onIndexedCount\":0,\"activeThreads\":0}}Content-Length: 179
\n
\n{\"jsonrpc\":\"2.0\",\"method\":\"$cquery/progress\",\"params\":{\"indexRequestCount\":0,\"doIdMapCount\":0,\"loadPreviousIndexCount\":0,\"onIdMappedCount\":0,\"onIndexedCount\":0,\"activeThreads\":0}}Content-Length: 311
\n
\n{\"jsonrpc\":\"2.0\",\"method\":\"textDocument/publishDiagnostics\",\"params\":{\"uri\":\"file:///home/rietmann/drz/c++_test/src/testquick.cpp\",\"diagnostics\":[{\"range\":{\"start\":{\"line\":4,\"character\":6},\"end\":{\"line\":4,\"character\":6}},\"severity\":2,\"source\":\"cquery\",\"message\":\"unused variable 'steve' [-Wunused-variable]\"}]}}")

**Comment by rietmann** [comment]
For my setup, I report this as fixed with cquery version 20180122.1922.

Not sure what changed, but thanks!

Max
